### PR TITLE
Add appender for JSON access logs

### PR DESCRIPTION
**Pull request by philandstuff**
This adds an appender specifically for access logs.  It allows you to
have request logs emitted in JSON format.  To use it you add the
following to your dropwizard config:

    server:
      requestLog:
        - type: access-logstash-console

This will only work if you are using logback-access style request logs
-- this is the default, but it may be disabled if you have
`type: classic` in your requestLog configuration.

This won't change existing behaviour; it only adds the new appender.

I've only added console logging for the moment because I'm
specifically targetting apps in a PaaS which will log to console.  We
can add File or Syslog appenders later if there's a demonstrated need.

### CODE REVIEW

File: src/main/java/uk/gov/ida/dropwizard/logstash/AccessLogstashConsoleAppenderFactory.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                                         LevelFilterFactory<IAccessEvent> levelFilterFactory,
+                                         AsyncAppenderFactory<IAccessEvent> asyncAppenderFactory) {
+
+        LogstashAccessEncoder encoder = new LogstashAccessEncoder();
+        encoder.setFieldNames(logstash675FieldNames);
+        encoder.setContext(context);
+        encoder.start();
+
+        final ConsoleAppender<IAccessEvent> appender = new ConsoleAppender<>();
+        appender.setName("logstash-console-appender");
```
Comment (willp-bl) [inline src/main/java/uk/gov/ida/dropwizard/logstash/AccessLogstashConsoleAppenderFactory.java]: can this be access-logstash-console, too

**Comment by willp-bl** [comment]
It would be good to have some test coverage for the new appender, we've added dropwizard apprule tests for the others, so it will mostly be a copy of the existing LogstashConsoleAppenderAppRuleTest

**Comment by willp-bl** [comment]
Also, could you update the README with details of how to use the new appender?

**Comment by philandstuff** [comment]
@willp-bl updated

**Comment by philandstuff** [comment]
urgh, fixing failing tests. they interfere with each other due to statically overriding System.out

**Comment by philandstuff** [comment]
@willp-bl fixed now

**Comment by willp-bl** [comment]
Looks good to me 👍

**Pull request merged**